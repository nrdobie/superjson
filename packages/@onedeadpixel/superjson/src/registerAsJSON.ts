/**
 * @module @onedeadpixel/superjson
 */
import SJSON from '.'

/**
 * Original JSON object
 */
export const originalJSON = global.JSON

/**
 * Makes the default SuperJSON instance what is used instead of JSON
 */
export function registerAsJSON() {
  // @ts-ignore
  global.JSON = SJSON
}

/**
 * Restores JSON back to its default
 */
export function unregisterAsJSON() {
  global.JSON = originalJSON
}
