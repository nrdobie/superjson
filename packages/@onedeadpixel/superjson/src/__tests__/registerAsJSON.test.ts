import { registerAsJSON, unregisterAsJSON } from '../registerAsJSON'

afterEach(() => {
  unregisterAsJSON()
})

test('registerAsJSON should replace JSON with SJSON', () => {
  registerAsJSON()

  expect(JSON.stringify(new Date('1902-06-24T00:00:00Z'))).toMatchSnapshot()
})

test('unregisterAsJSON should replace SJSON with JSON', () => {
  registerAsJSON()

  expect(JSON.stringify(new Date('1902-06-24T00:00:00Z'))).toMatchSnapshot()

  unregisterAsJSON()

  expect(JSON.stringify(new Date('1902-06-24T00:00:00Z'))).toMatchSnapshot()
})
