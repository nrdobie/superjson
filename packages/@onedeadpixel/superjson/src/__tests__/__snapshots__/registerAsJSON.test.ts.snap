// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registerAsJSON should replace JSON with SJSON 1`] = `"{\\"__sj_type\\":\\"Date\\",\\"__sj_value\\":\\"1902-06-24T00:00:00.000Z\\"}"`;

exports[`unregisterAsJSON should replace SJSON with JSON 1`] = `"{\\"__sj_type\\":\\"Date\\",\\"__sj_value\\":\\"1902-06-24T00:00:00.000Z\\"}"`;

exports[`unregisterAsJSON should replace SJSON with JSON 2`] = `"\\"1902-06-24T00:00:00.000Z\\""`;
